### PR TITLE
[SPARK-19682][SparkR] Issue warning (or error) when subset method "[[" takes vector index

### PR DESCRIPTION
**Pull request by actuaryzhang**
## What changes were proposed in this pull request?
The `[[` method is supposed to take a single index and return a column. This is different from base R which takes a vector index.  We should check for this and issue warning or error when vector index is supplied (which is very likely given the behavior in base R). 

Currently I'm issuing a warning message and just take the first element of the vector index. We could change this to an error it that's better. 

## How was this patch tested?
new tests

### CODE REVIEW

**Comment by actuaryzhang** [comment]
@felixcheung 
Simple example to illustrate this
```
df <- suppressWarnings(createDataFrame(iris))
df[[1:2]]
```
Instead of issuing warning and taking the first element, we can throw a more meaningful error message. Minor issue, but many R users may be accustomed to use vector index with `[[`.

**Comment by felixcheung** [comment]
Jenkins, retest this please

**Comment by felixcheung** [comment]
That's one weird test failure.
Looks good

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -1776,6 +1780,10 @@ setMethod("[[", signature(x = "SparkDataFrame", i = "numericOrcharacter"),
 #' @note [[<- since 2.1.1
 setMethod("[[<-", signature(x = "SparkDataFrame", i = "numericOrcharacter"),
           function(x, i, value) {
+            if (length(i) > 1) {
+              warning("Subset index has length > 1. Only the first index is used.")
+              i <- i[1]
+            }
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: actually, sorry I missed this - could you add a test for this?

**Review by felixcheung — CHANGES_REQUESTED** [review]

File: R/pkg/R/DataFrame.R (hunk 1 above)
Comment (actuaryzhang) [inline R/pkg/R/DataFrame.R]: @felixcheung Test added. Thanks for catching this.

**Comment by felixcheung** [comment]
thanks - merged to master and branch-2.1

**Pull request closed without merging**